lib/customer.rb

```diff
@@ -0,0 +1,38 @@
+require "csv"
+require "awesome_print"
+
+class Customer
+  attr_accessor :email, :address
+  attr_reader :id
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    customer_data_array = []
+    customer_data_collection = []
+    CSV.open("../data/customers.csv", "r").each do |row|
```

Comment:
Quick note on path conventions! We assume that we are calling these files from the root of the directory, not from lib. This will make more sense once we get to rails, but just keep that in mind: when you run rake, run it from the root of the project. The path we recommend is `data/customers.csv`